File: Java/IntegerListFromUsp/src/Exercicio12.java
import java.util.Scanner;

public class Exercicio12 {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        scanner.reset();

        System.out.print("Digite um número: ");
        int numA = scanner.nextInt();

        System.out.print("Digite outro número: ");
        int numB = scanner.nextInt();

        int resto = 0;

        while (numA != 0 && numB != 0) {
            resto = numA % numB;
            numA = numB;
            numB = resto;
        }

        System.out.println("MDC é %d".formatted(numA));
    }

}

File: CPP/IntegerListFromUsp/src/Exercicio12.cpp
#include <iostream>

using namespace std;

int main() {
    int a, b;
    cout << "Digite dois inteiros positivos: ";
    cin >> a >> b;

    int original_a = a;
    int original_b = b;

    while (b != 0) {
        int resto = a % b;
        a = b;
        b = resto;
    }

    cout << "MDC(" << original_a << ", " << original_b << ") = " << a << endl;
    return 0;
}
